src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -93,9 +93,9 @@ public static PackageSpec GetPackageSpec(JObject rawPackageSpec, string name, st
             {
                 try
                 {
-                    bool hasVersionSnapshot;
-                    packageSpec.Version = SpecifySnapshot(version.Value<string>(), snapshotValue, out hasVersionSnapshot);
-                    packageSpec.HasVersionSnapshot = hasVersionSnapshot;
+                    var versionString = version.Value<string>();
+                    packageSpec.HasVersionSnapshot = PackageSpecUtility.IsSnapshotVersion(versionString);
```

Comment:
Should we really support this? Like, this was a `preview2` thing. We should have a design discussion about this given that `dotnet migrate` strips `-*` and `dotnet build(3)` does not have this logic at all.

It think it would be safer to start with not having snapshot versions supported and, only after a fully thought through design is complete, add in snapshot versions.